certora/specs/SocialRecoveryModule.spec

```diff
@@ -302,3 +302,49 @@ rule disabledRecoveryModuleResultsInFinalizationRevert(env e) {
             safeContract.getOwners().length == 1 &&
             currentThreshold == safeContract.getThreshold());
 }
+
+// Recovery can be cancelled
+rule cancelRecovery() {
+    env e;
+    require e.msg.sender == safeContract;
+    require e.msg.value == 0;
+
+    // A recovery request must be initiated.
+    require currentContract.recoveryRequests[safeContract].executeAfter > 0;
+    require currentContract.walletsNonces[safeContract] > 0;
+
+    currentContract.cancelRecovery@withrevert(e, safeContract);
+    assert !lastReverted;
+}
+
+// Cancelling recovery for a wallet does not affect other wallets
+rule cancelRecoveryDoesNotAffectOtherWallet() {
+    env e;
+    address otherWallet;
+    require otherWallet != safeContract;
```

Comment:
We can have an imply instead of this, so if `otherWallet != safeContract` we only make those checks then.

Reply:
I also prefer implies as it "hides" less paths.

Reply:
Then I will keep assertions as they are.